docs/modeling/changes_for_4.rst

```diff
@@ -1,4 +1,181 @@
-Modeling Changes in astropy v.4.0
-*********************************
+.. include:: links.inc
 
-(placeholder)
\ No newline at end of file
+**************************
+Changes to Modeling in 4.0
+**************************
+
+
+In order to make the internal code simpler, improve performance, and
+make the behavior of parameters in compound models more intuitive,
+many changes have been made internally to the modeling code,
+particularly to compound models and parameters. This page
+summarizes the important changes. More technical details are 
+given at the end, but it is generally not necessary to read those
+unless one wants to understand why some changes were necessary.
+
+- Support for expressions of modeling classes has been removed.
+  Expressions of model class instances are still fully supported.
+  This was done to simplify the implemenation, improve performance,
+  and support the new parameter semantics. For example:
+
+  No longer works::
+
+        NewCompoundClass = Gaussian1D + Const1D
+
+  Still works::
+
+        newmodel = Gaussian1D(3.2, 7.1, 2.1) + Const1D(3.)
+
+- Previous to 4.0, parameters were class descriptors, which meant
+  that they could not hold values for the models. Instead, the 
+  values were held inside the models. This resulted in confusion
+  when compound models were used since this necessitated that
+  the compound models make copies of the values. As a result, 
+  changing the value in the compound model did not change the
+  constituent model's parameter value and visa versa. Now
+  parameters are distinct instances for each use, and they
+  do now hold the value of the parameter. So compound models
+  now share the same values as the constituent models.
+
+- Previously when model sets were used, the parameter shape
+  did not show the corresponding dimension for the number 
+  of models. Now it does. For example:
+
+  Old::
+
+        In [1]: g = Gaussian1D([1,1], [1,2], [2,4], n_models=2)
+        In [2]: g.amplitude
+        Out[2]: Parameter('amplitude', value=[1. 1.])
+        In [3]: g.amplitude.shape
+        Out[3]: ()
+
+  New::
+
+        In [1]: g = Gaussian1D([1,1], [1,2], [2,4], n_models=2)
+        In [2]: g.amplitude
+        Out[2]: Parameter('amplitude', value=[1. 1.])
+        In [3]: g.amplitude.shape
+        Out[3]: (2,)
+
+- Previously the values were held in an array within the model
+  instance and it was possible to assign values to slices of
+  that array. Reassigning the array does update the parameters,
+  but assigning slices does not. The new approach is to either
+  replace the whole array by assigning to the parameters property
+  or assign directly to the parameter value.
+
+- The use of 'imputed' units, i.e., supplying input/output units
+  to a compound model without them but where the component models 
+  support the _parameter_units_for_data_units() method is much more
```

Comment:
```suggestion
  support the _parameter_units_for_data_units() method) is much more
```

Reply:
Can you try rephrasing this sentence for clarity? Perhaps putting the i.e. material in its own sentence would help?